api4/status.go

```diff
@@ -59,3 +59,35 @@ func getUserStatusesByIds(c *Context, w http.ResponseWriter, r *http.Request) {
 		w.Write([]byte(model.StatusListToJson(statusMap)))
 	}
 }
+
+func updateUserStatus(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireUserId()
+	if c.Err != nil {
+		return
+	}
+
+	status := model.StatusFromJson(r.Body)
+	if status == nil {
+		c.SetInvalidParam("status")
+		return
+	}
+
+	if !app.SessionHasPermissionToUser(c.Session, c.Params.UserId) {
+		c.SetPermissionError(model.PERMISSION_EDIT_OTHER_USERS)
+		return
+	}
+
+	switch status.Status {
+	case "online":
+		app.SetStatusOnline(c.Params.UserId, c.Session.Id, true)
```

Comment:
Can we pass a blank string for the session ID here? You'll need to update `app.SetStatusOnline` to not update the Session in the DB if `sessionId == ""` as well

Otherwise we might be updating the system admin's session when they are trying to update the status of another user